lib/listener.js

```diff
@@ -410,18 +414,27 @@ internals.Subscribers.prototype.add = function (socket, path) {
     else {
         this._items[socket.id] = { socket: socket, paths: [path] };
     }
+
+    if (this._onSubscribe) {
+        this._onSubscribe(socket, path);
+    }
 };
 
 
 internals.Subscribers.prototype.remove = function (socket, path) {
 
-    if (!path) {
-        delete this._items[socket.id];
+    var item = this._items[socket.id];
+    if (!item) {
         return;
     }
 
-    var item = this._items[socket.id];
-    if (!item) {
+    if (!path) {
+        if (this._onUnsubscribe) {
+            this._onUnsubscribe(socket);
```

Comment:
I don't like this inconsistent behavior. We should be able to find the paths and provide them, maybe as an array or multiple calls?